From 11b374d201af6c3e75026b72dd9a7a19ccdc9c0c Mon Sep 17 00:00:00 2001
From: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
Date: Thu, 19 Sep 2024 09:54:43 +0200
Subject: [PATCH 3/3] Cargo: reduce binary size for release profile

In the release profile of Cargo.toml, add `codegen-units = 1`,
`strip = true`, and remove `debug = true`, to reduce binary size of
afterburn included in production images.

Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
---
 Cargo.toml | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 7cd4b91..34001cb 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,8 +30,10 @@ name = "afterburn"
 path = "src/main.rs"
 
 [profile.release]
-# We assume we're being delivered via e.g. RPM which supports split debuginfo
-debug = true
+# Flatcar: Reduce binary size for prod images by using one codegen unit,
+# stripping binary, and removing debug = true.
+codegen-units = 1
+strip = true
 
 [features]
 cl-legacy = []
-- 
2.44.2

